src/main/java/uk/gov/hmcts/reform/idam/client/IdamClient.java

```diff
@@ -47,8 +48,9 @@ public UserDetails getUserDetails(String bearerToken) {
         return idamApi.retrieveUserDetails(bearerToken);
     }
 
-    public String getAccessToken(String username, String password) {
-        TokenRequest tokenRequest =
+    // when using the acess token you may need to add "Bearer "
+    public TokenResponse getAccessTokenResponse(String username, String password) {
```

Comment:
shouldn't this library handle caching? I assume that's what you want from this?

Reply:
we handle caching in services.

Reply:
why not cache here though? that way all consumers don't need to implement and its handled out of the box

Reply:
because we invalidate the cache value depending on ccd response, this api does not know anything about ccd. Later we will try to make it separate library.